configure.ac

```diff
@@ -5822,13 +5823,20 @@ AC_ARG_WITH(
     AS_CASE([$with_readline],
       [editline|edit], [with_readline=edit],
       [yes|readline], [with_readline=readline],
-      [no], [],
+      [no], [with_readline=default],
       [AC_MSG_ERROR([proper usage is --with(out)-readline@<:@=editline|readline|no@:>@])]
     )
   ],
-  [with_readline=readline]
+  [with_readline=default]
 )
 
+dnl gh-105323: Need to handle the macOS editline as an alias of readline.
+AS_VAR_IF([with_readline], [default], [
+  AS_CASE([$ac_sys_system/$ac_sys_release],
+    [Darwin/*], [AC_DEFINE([WITH_APPLE_READLINE], [1]) with_readline=readline],
+    [with_readline=readline])
```

Comment:
You set `with_readline=readline` no matter which branch; perhaps you can put that assignment outside the switch.